protocol/x/clob/keeper/match_state.go

```diff
@@ -24,17 +59,26 @@ func (k Keeper) GetLastTradePriceForPerpetual(
 	return types.Subticks(result.Value), true
 }
 
-// SetLastTradePriceForPerpetual sets the last trade price for a perpetual.
-func (k Keeper) SetLastTradePriceForPerpetual(
+// SetTradePricesForPerpetual sets the trade prices for a perpetual.
+func (k Keeper) SetTradePricesForPerpetual(
```

Comment:
For my understanding, where do you expect to be calling this in order to set the trade prices? In the memclob matching loop or do we populate this from the operations queue?

Reply:
in process_operations when we actually process these matches.